R/getTimeDivergence.R

```diff
@@ -58,21 +60,24 @@ getTimeDivergence <- function(se, sample_field, sample_id, time_field, time_inte
     mat[, time_field][location] <- extracted_time
 
     #related samples are chosen from assay
-    x <- as.matrix(assay(tse)[,rownames(sample_df)[which(is.na(mat[, time_field]) == FALSE)]])
+    x <- as.matrix(assay(se)[,rownames(sample_df)[which(is.na(mat[, time_field]) == FALSE)]])
 
     if(!transposed){
       x <- t(x)
     }
 
-    timedivergence_n <- as.matrix(vegan::vegdist(x), "bray")
+    #timedivergence_n <- calculateDistance(x)
+    timedivergence_n <- calculateJSD(x)
 
```

Comment:
Should `calculateJDS` and `calculateDistance ` be used here in the same function? If yes, then an optional parameter can be placed to choose which one to use and there could be a default function as well. The other function `calculateUniFrac` works differently and cannot be used in the same way as others. I am trying to find different ways to make it applicable in this function.

Reply:
Yes, ideally we could choose use all of these common beta diversity methods also with time series.

But already a proper implementation with `calculateDistance` would be a valuable addition. Can start with that if adding others is too complex in the first stage. Another option is that we have separate functions for these different beta diversities, with uniform naming style.